### PR TITLE
Use the association primary key when importing

### PR DESCRIPTION
**Pull request by dpogue**
When importing into an association, it was using the owner class table primary key (usually `id`) rather than the `:primary_key` option specified on the association.

### CODE REVIEW

**Comment by jkowens** [comment]
Thanks for this! Any chance you could add test coverage for this case?

**Comment by dpogue** [comment]
Ah, will do!

**Comment by jkowens** [comment]
It's strange that the test is failing for Rails 3.2, but can you move this test to the `test/import_test.rb` file? The file it is currently in is specific testing `on_duplicate_key_update`.

File: test/support/shared_examples/on_duplicate_key_update.rb [hunk 1]
```diff
@@ -284,6 +284,27 @@ def should_support_basic_on_duplicate_key_update
...
+        let(:update_columns) { [:token] }
+
+        macro(:perform_import) do |*opts|
+          @user.user_tokens.import columns, values, opts.extract_options!.merge(on_duplicate_key_update: update_columns, validate: false)
+        end
+        macro(:created_token) { UserToken.find(5) }
+
+        setup do
+          User.import [User.new(id: 1, name: 'Solomon')], validate: false
+          @user = User.find 1
```
Comment (jkowens) [inline test/support/shared_examples/on_duplicate_key_update.rb]: This could be simplified a bit:

` @user = User.create(id: 1, name: 'Solomon')`

**Comment by dpogue** [comment]
Thanks! I've moved the test and it's all passing now :)

**Comment by jkowens** [comment]
Looks good, thanks again 👍

**Comment by jkowens** [comment]
Released in v0.24.0 👍

**Pull request merged**